azure/services/privatedns/client.go

```diff
@@ -19,166 +19,13 @@ package privatedns
-	if err != nil {
-		return err
-	}
-	err = future.WaitForCompletionRef(ctx, ac.privatezones.Client)
-	if err != nil {
-		return err
-	}
-	_, err = future.Result(ac.privatezones)
-	return err
-}
-
-// GetLink returns a vnet link.
-func (ac *azureClient) GetLink(ctx context.Context, resourceGroupName, zoneName, vnetLinkName string) (privatedns.VirtualNetworkLink, error) {
-	ctx, _, done := tele.StartSpanWithLogger(ctx, "privatedns.AzureClient.GetLink")
-	defer done()
-	vnetLink, err := ac.vnetlinks.Get(ctx, resourceGroupName, zoneName, vnetLinkName)
-	if err != nil {
-		return privatedns.VirtualNetworkLink{}, err
-	}
-	return vnetLink, nil
-}
-
-// CreateOrUpdateLink creates or updates a virtual network link to the specified Private DNS zone.
-func (ac *azureClient) CreateOrUpdateLink(ctx context.Context, resourceGroupName, privateZoneName, name string, link privatedns.VirtualNetworkLink) error {
-	ctx, _, done := tele.StartSpanWithLogger(ctx, "privatedns.AzureClient.CreateOrUpdateLink")
-	defer done()
-
-	future, err := ac.vnetlinks.CreateOrUpdate(ctx, resourceGroupName, privateZoneName, name, link, "", "")
-	if err != nil {
-		return err
-	}
-	err = future.WaitForCompletionRef(ctx, ac.vnetlinks.Client)
-	if err != nil {
-		return err
-	}
-	_, err = future.Result(ac.vnetlinks)
-	return err
-}
-
-// DeleteLink deletes a virtual network link to the specified Private DNS zone.
-func (ac *azureClient) DeleteLink(ctx context.Context, resourceGroupName, privateZoneName, name string) error {
-	ctx, _, done := tele.StartSpanWithLogger(ctx, "privatedns.AzureClient.DeleteLink")
-	defer done()
-
-	future, err := ac.vnetlinks.Delete(ctx, resourceGroupName, privateZoneName, name, "")
-	if err != nil {
-		return err
-	}
-	err = future.WaitForCompletionRef(ctx, ac.vnetlinks.Client)
-	if err != nil {
-		return err
-	}
-	_, err = future.Result(ac.vnetlinks)
-	return err
-}
-
-// CreateOrUpdateRecordSet creates or updates a record set within the specified Private DNS zone.
-func (ac *azureClient) CreateOrUpdateRecordSet(ctx context.Context, resourceGroupName string, privateZoneName string, recordType privatedns.RecordType, name string, set privatedns.RecordSet) error {
-	ctx, _, done := tele.StartSpanWithLogger(ctx, "privatedns.AzureClient.CreateOrUpdateRecordSet")
-	defer done()
-
-	_, err := ac.recordsets.CreateOrUpdate(ctx, resourceGroupName, privateZoneName, recordType, name, set, "", "")
-	return err
-}
-
-// DeleteRecordSet deletes a record set within the specified Private DNS zone.
-func (ac *azureClient) DeleteRecordSet(ctx context.Context, resourceGroupName string, privateZoneName string, recordType privatedns.RecordType, name string) error {
-	ctx, _, done := tele.StartSpanWithLogger(ctx, "privatedns.AzureClient.DeleteRecordSet")
-	defer done()
-
-	_, err := ac.recordsets.Delete(ctx, resourceGroupName, privateZoneName, recordType, name, "")
-	return err
-}
+var _ Getter = (*azureZonesClient)(nil)
+var _ Getter = (*azureVirtualNetworkLinksClient)(nil)
```

Comment:
Don't we need to enforce type checks for `azureRecordsClient` also like we did for `azureZonesClient ` and `azureVirtualNetworkLinksClient ` ?

Reply:
Now that we are using `async.Getter`, we won't need this file anymore.